packages/ui/src/components/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+    ArrowUpRightIcon,
+    Bars3BottomRightIcon,
+    MagnifyingGlassIcon,
+    XMarkIcon,
+} from '@heroicons/react/24/outline';
+import type {
+    Category,
+    Culture,
+    ExtendedStory,
+    NewsroomLanguageSettings,
+    UploadedImage,
+} from '@prezly/sdk';
+import Image from '@prezly/uploadcare-image';
+import Link from 'next/link';
+import { type MouseEvent as ReactMouseEvent, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { useDevice } from '@/hooks';
+
+import { Button } from '../Button';
+
+import { CategoriesDropdown, LanguagesDropdown } from './components';
+import type { NavigationLayout } from './types';
+import { extractDomainFromUrl } from './util';
+
+export interface Props {
+    className?: string;
+    categories: Category[];
+    languages: NewsroomLanguageSettings[];
+    layout?: NavigationLayout;
+    logo?: UploadedImage | null;
+    siteName: string;
+    publicGalleriesCount: number;
+    externalSiteLink?: string;
+    isSearchEnabled?: boolean;
```

Comment:
I believe that all of these properties can be pulled from `Newsroom` API object. That would help reduce the amount of props required.